### PR TITLE
Add mqtt and kafka systems

### PR DESCRIPTION
**Pull request by georific**
## Status
**READY**

## Description
Add the ability use mqtt or kafka

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
git fetch --all
git checkout <feature_branch> 
npm test
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: server/dev/user.clj [hunk 1]
```diff
@@ -17,9 +17,30 @@
...
 
+(defn start-mqtt-system []
+  (log/info "starting mqtt system")
+  (make-spacon-server-mqtt {:http-config {:env                     :dev
+                                          ::server/join?           false
+                                          ::server/allowed-origins {:creds true
+                                                                    :allowed-origins (constantly true)}}
+                            :mqtt-config {:broker-url (or (System/getenv "MQTT_BROKER_URL")
+                                                          "tcp://localhost:1883")}})
+  )
```
Comment (mrcnc) [inline server/dev/user.clj]: Probably want to move this paren to conform to the conventional style.  Try using [parinfer](https://shaunlebron.github.io/parinfer/index.html) and see how you like it.

File: server/project.clj [hunk 2]
```diff
@@ -4,6 +4,7 @@
   :license {:name "Apache License, Version 2.0"
             :url "https://www.apache.org/licenses/LICENSE-2.0"}
   :dependencies [[org.clojure/clojure "1.9.0-alpha14"]
+                 [org.clojure/core.async "0.3.442"]
```
Comment (mrcnc) [inline server/project.clj:7]: Not sure why you need this....isn't core.async in the 1.9 release of clojure?

Comment (tetriscode) [reply server/project.clj:7]: I added it. There are fixes to core.async wrt channel exceptions that are in that build. The release far more often than the main.

File: server/src/spacon/components/form/core.clj [hunk 3]
```diff
@@ -15,10 +15,10 @@
 (ns spacon.components.form.core
   (:require [com.stuartsierra.component :as component]
             [spacon.components.form.db :as formmodel]
-            [spacon.components.mqtt.core :as mqttapi]
+            [spacon.components.queue.protocol :as queueapi]
             [clojure.spec :as s]
             [clojure.spec.gen :as gen]
-            [spacon.components.mqtt.core :as mqttapi]
+            [spacon.components.kafka.core :as kafkaapi]
```
Comment (tetriscode) [inline server/src/spacon/components/form/core.clj]: This line can be removed

File: server/project.clj (hunk 2 above)
Comment (mrcnc) [reply server/project.clj:7]: 👍

File: server/project.clj [hunk 4]
```diff
@@ -30,11 +31,12 @@
                  [overtone/at-at "1.2.0"]
                  [clj-http "2.3.0"]
                  [com.gfredericks/test.chuck "0.2.7"]
-                 [jonase/eastwood "0.2.3" :exclusions [org.clojure/clojure]]
                  [com.draines/postal "2.0.2"]
                  [org.clojure/tools.logging "0.3.1"]
+                 [funcool/cats "2.0.0"]
                  [org.apache.kafka/kafka-clients "0.10.1.1"]
-                 [org.apache.kafka/kafka-streams "0.10.0.0-cp1" :exclusions [org.slf4j/slf4j-log4j12]]
+                 [org.apache.kafka/kafka-streams "0.10.0.0-cp1"
```
Comment (tetriscode) [inline server/project.clj:38]: Is there a 0.10.1 release of this lib instead of 0.10.0?

**Pull request merged**